src/app/clusters/commodity-tariff-server/CommodityTariffContainers.h

```diff
@@ -0,0 +1,357 @@
+/**
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cassert>
+#include <pw_containers/algorithm.h>
+#include <pw_containers/flat_map.h>
+#include <pw_containers/vector.h>
+
+#include "lib/core/CHIPError.h"
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+
+#include "CommodityTariffConsts.h"
+
+namespace chip {
+namespace app {
+namespace CommodityTariffContainers {
+
+template <typename T, size_t kMaxSize>
+class CTC_UnorderedSet
```

Comment:
Pigweed has some unordered set and map classes; is there some missing functionality there that prevents you from using it directly?